core/src/toga/app.py

```diff
@@ -281,7 +262,22 @@ def __init__(
         :param description: A brief (one line) description of the app. If not provided,
             the metadata key ``Summary`` will be used.
         :param startup: A callable to run before starting the app.
+        :param on_exit: The handler to invoke before the application exits.
+        :param windows: **DEPRECATED** – Windows are now automatically added to the
+            current app. Passing this argument will cause an exception.
         """
+        ######################################################################
+        # 2023-10: Backwards compatibility
+        ######################################################################
+        if windows is not None:
+            raise ValueError(
+                "The `windows` constructor argument of toga.App has been removed. "
+                "Windows are now automatically added to the current app."
+            )
```

Comment:
As part of the Window audit, I changed it so that Windows are added to the App when they're created, instead of when they're shown. Then as part of this PR, I added a check to make sure that the App is created before the Window:

https://github.com/beeware/toga/blob/ff64a99fbdb8af5035b70068cfefa2f0711a567f/core/src/toga/window.py#L134-L137

I appreciate that this may break some existing code, but I think it's worth it because it makes it significantly easier to reason about the relationship between App and Window, both for us and for the users. Anyone who's been following the style of the Toga examples will not be affected.

This means that any windows passed to the App constructor must already have been added to a different App. so there's no way we can handle that situation sensibly.